@xen-orchestra/lite/src/components/AddNetwork.tsx

```diff
@@ -0,0 +1,295 @@
+import AddIcon from '@mui/icons-material/Add'
+import React from 'react'
+import SettingsBackupRestoreIcon from '@mui/icons-material/SettingsBackupRestore'
+import { Map } from 'immutable'
+import { SelectChangeEvent } from '@mui/material'
+import { withState } from 'reaclette'
+
+import ActionButton from './ActionButton'
+import Button from './Button'
+import Checkbox from './Checkbox'
+import Input from './Input'
+import IntlMessage from './IntlMessage'
+import Select from './Select'
+import { alert } from './Modal'
+
+import XapiConnection, { BOND_MODE, ObjectsByType, Pif, PifMetrics } from '../libs/xapi'
+
+interface ParentState {
+  objectsByType: ObjectsByType
+  objectsFetched: boolean
+  xapi: XapiConnection
+}
+
+interface State {
+  form: {
+    [key: string]: unknown
+    bondMode: string
+    description: string
+    isBonded: boolean
+    isEmptyBondMode: boolean
+    isEmptyLabel: boolean
+    isInsufficientNumberOfInterfaces: boolean
+    mtu: string
+    nameLabel: string
+    pifsId: string[]
+    vlan: string
+  }
+}
+
+interface Props {}
+
+interface ParentEffects {}
+
+interface Effects {
+  createNetwork: () => Promise<void>
+  handleChange: (e: SelectChangeEvent<unknown> | React.ChangeEvent<{ name: string; value: unknown }>) => void
+  resetForm: () => void
+  toggleBonded: () => void
+}
+
+interface Computed {
+  pifsCollection?: Pif[]
+  pifs?: Map<string, Pif>
+  pifsMetrics?: Map<string, PifMetrics>
+}
+
+const BUTTON_STYLES = {
+  marginRight: '1em',
+  width: 'fit-content',
+}
+
+const OPTION_PIF_RENDERER = (pif: Pif, { pifsMetrics }: { pifsMetrics: Computed['pifsMetrics'] }) =>
+  `${pif.device} (${pifsMetrics?.find(metrics => metrics.$ref === pif.metrics)?.device_name ?? 'unknown'})`
+
+const INPUT_STYLES = {
+  marginBottom: '1em',
+  width: '100%',
+}
+
+const getInitialFormState = (): State['form'] => ({
+  bondMode: '',
+  description: '',
+  isBonded: false,
+  isEmptyBondMode: false,
+  isEmptyLabel: false,
+  isInsufficientNumberOfInterfaces: false,
+  mtu: '',
+  nameLabel: '',
+  pifsId: [],
+  vlan: '',
+})
+
+const AddNetwork = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      form: getInitialFormState(),
+    }),
+    computed: {
+      pifs: state => state.objectsByType.get('PIF'),
+      pifsMetrics: state => state.objectsByType.get('PIF_metrics'),
+      pifsCollection: state =>
```

Comment:
`filteredPifs`? We already have a collection called `pifs`. When you read both names, you need to be able to understand the difference between those 2 collections. With `pifs`/`pifsCollection`, I still don't understand what the difference is supposed to be. With `pifs`/`filteredPifs`, I understand that we filtered `pifs` to get `filteredPifs`.